RBS.Auth.WebApi/RBS.Auth.WebApi/Startup.cs

```diff
@@ -1,84 +1,85 @@
 using RBS.Auth.Common;
+using RBS.Auth.Db;
 using RBS.Auth.Services.Authenticate;
+using RBS.Auth.Services.Hashing;
 using RBS.Auth.Services.Interfaces.Authenticate;
+using RBS.Auth.Services.Interfaces.Hashing;
 using RBS.Auth.Services.Interfaces.Tokens;
 using RBS.Auth.Services.Tokens;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
 
-namespace RBS.Auth.WebApi
+namespace RBS.Auth.WebApi;
+
+public class Startup
 {
-    public class Startup
+    public Startup(IConfiguration configuration, IWebHostEnvironment env)
     {
-        public Startup(IConfiguration configuration, IWebHostEnvironment env)
-        {
-            Configuration = configuration;
-            Environment = env;
-        }
+        var builder = new ConfigurationBuilder()
+            .SetBasePath(env.ContentRootPath)
+            .AddJsonFile("appsettings.json", false, true)
+            .AddJsonFile($"appsettings.{env.EnvironmentName}.json", true, true)
+            .AddEnvironmentVariables();
 
-        public IConfiguration Configuration { get; }
-        public IWebHostEnvironment Environment { get; }
+        Configuration = builder.Build();
+        Environment = env;
+    }
 
-        // This method gets called by the runtime. Use this method to add services to the container.
-        public void ConfigureServices(IServiceCollection services)
-        {
-            ConfigureInternalServices(services);
+    public IConfiguration Configuration { get; }
+    public IWebHostEnvironment Environment { get; }
+
+    // This method gets called by the runtime. Use this method to add services to the container.
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.ConfigureDatabase(Configuration);
+
+        services.AddAutoMapper(typeof(Startup));
+
+        ConfigureInternalServices(services);
 
-            services.AddControllers();
+        services.AddControllers();
 
-            services.Configure<AuthOptions>(Configuration.GetSection("Auth"));
+        services.Configure<AuthOptions>(Configuration.GetSection("Auth"));
 
-            if (!Environment.IsProduction())
+        if (!Environment.IsProduction())
+            services.AddSwaggerGen(options =>
             {
-                services.AddSwaggerGen(options =>
-                {
-                    options.SwaggerDoc("v1", new OpenApiInfo { Title = "RBS.Auth.WebApi", Version = "v1" });
-                });
-            }
-        }
+                options.SwaggerDoc("v1", new OpenApiInfo {Title = "RBS.Auth.WebApi", Version = "v1"});
+            });
+    }
 
-        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
```

Comment:
Consider to add some editor config, in order to avoid "code format chages"

Reply:
Added